pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -671,8 +672,19 @@ public int getConcurrency() throws SQLException {
       }
     } else {
       // string
-      if (oid == Oid.TIMESTAMPTZ || oid == Oid.TIMESTAMP || oid == Oid.TIMETZ) {
-        return getTimestampUtils().toOffsetDateTime(castNonNull(getString(i)), oid != Oid.TIMETZ);
+
+      if (oid == Oid.TIMESTAMPTZ || oid == Oid.TIMESTAMP )  {
+
+        OffsetDateTime offsetDateTime = getTimestampUtils().toOffsetDateTime(castNonNull(getString(i)));
+        if ( offsetDateTime != OffsetDateTime.MAX && offsetDateTime != OffsetDateTime.MIN ) {
+          return offsetDateTime.withOffsetSameInstant(ZoneOffset.UTC);
+        } else {
+          return offsetDateTime;
+        }
+
+      }
+      if ( oid == Oid.TIMETZ ) {
+        return getTimestampUtils().toOffsetDateTime(castNonNull(getString(i)));
```

Comment:
I first thought that we may need the atDate(LOCAL_DATE_EPOCH) here. But its not needed because a ParsedDatestamp I stance defaults to epoch, too: https://github.com/davecramer/pgjdbc/blob/28e3434c5f751f54bd49e0a2c2a88dff45524222/pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java#L165

Reply:
are you suggesting a change here or just commenting on the existing code ?

Reply:
Hi, this was just a note to the reader (but could possibly added asa comment). The TIMETZ value is just a time in string format. But as above link tells you, the `toOffsetDateTime(String)` adds the EPOCH already, because the year/month/day values in ParsedTimestamp default to 1970, 1, 1, so when we convert the ParsedDatestamp instance to `OffsetDateTime` it get the default values. I was just unsure about this when reiewing the patch.

Sorry for the comment, it was just informational.